src/query/api/v1/handler/types.go

```diff
@@ -29,4 +29,7 @@ const (
 
 	// RoutePrefixV1 is the v1 prefix for all coordinator routes
 	RoutePrefixV1 = "/api/v1"
+
+	// LimitHeader is the header added when series limits are exceeded.
+	LimitHeader = "Series-Limit-Exceeded"
```

Comment:
Could we make this a more generic header name, which the value of we could concat? (for instance to describe limit applied for max fetch series as well as region not being available).

Really want to describe is a partial response, and why (and if multi reasons, then each one) it was a partial response.

Something generic like "M3-Results-Limited" with a comma separated string value of reasons seems like a reasonable start.  I proposed embedded JSON to get fancy, but that's probably too far.  If we ever need more information we can put it in a separate header.  Just a comma separated set of string enum like values probably works for now with a single header.

Here's the conversation I had:
> chengyu [1:56 PM]
> Is there a way to surface this information back? Our service wants to adjust our query pattern on excessive queries.
> Otherwise, we have no way to know if there is no timeseries or we are hitting some limit somewhere..
> 
> Rob Skillington [1:59 PM]
> yeah I’m talking to artem right now, we might just add a `M3-Results-Limited-QueryLimit: true` header
> along with a `M3-Results-Limited-Any: true`
> because if you do a fanout to other regions and those regions are down
> that’s another different type of limit
> 
> chengyu [2:01 PM]
> What about just “M3-Results-Limited” and the value be the actual reason?
> Instead of having multiple different type of headers that we probably need to adapt later (edited)

Reply:
Gonna tweak how we do this a bit, hopefully get some better results out